compiler/internal/analysis/info.go

```diff
@@ -211,18 +251,30 @@ type FuncInfo struct {
 	returnStmts []astPath
 	// List of other named functions from the current package this function calls.
 	// If any of them are blocking, this function will become blocking too.
-	localNamedCallees map[*types.Func][]astPath
+	localInstCallees *typeparams.InstanceMap[[]astPath]
 	// List of function literals directly called from this function (for example:
 	// `func() { /* do stuff */ }()`). This is distinct from function literals
 	// assigned to named variables (for example: `doStuff := func() {};
 	// doStuff()`), which are handled by localNamedCallees. If any of them are
 	// identified as blocking, this function will become blocking too.
 	literalFuncCallees map[*ast.FuncLit][]astPath
+	// resolver is used by this function instance to resolve any type arguments
+	// for internal function calls.
+	// This may be nil if not an instance of a generic function.
+	resolver *typeparams.Resolver
 
 	pkgInfo      *Info // Function's parent package.
 	visitorStack astPath
 }
 
+// HasBlocking indicates if this function may block goroutine execution.
+//
+// For example, a channel operation in a function or a call to another
+// possibly blocking function may block the function.
+func (fi *FuncInfo) HasBlocking() bool {
```

Comment:
Elsewhere we use `IsBlocking`. Should this be the same for consistency, or is there a meaningful difference in the naming convention?

Reply:
That's a good question. I was replacing the repeat code `len(fi.Blocking) != 0` in a bunch of places so I thought, "this is checking if the function info _has blocking_ nodes", but now I'm wondering if that means "the function _is blocking_ because it has blocking nodes."

For consistency, I'll change it to `IsBlocking`